### ISSUE TITLE
dnbc4tools rna run: Unable to find custom marker genes in the 02.count/features.tsv.gz file even they are appeared in the 01.data/final_sorted.bam

### ISSUE THREAD

**Issue by zebrafishCC** [MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software#74]
Dear Shuangshuang, 
      I created a  custom reference database by adding fluorescence genes (gtf, mcherry, and bfp)  sequence and annotation information into the reference human genome. I then ran the 'dnbc4tools  rna run' command to get the count matrix using my dnbc4 fastq reads. I could ran the command successfully and obtained count matrix, however, I could not find my custom genes. Ridiculously, I found that some reads have aligned to gtf gene region by checking the 01.data/final_sorted.bam files (as shown by the following figures), and Gtf fluorescence have been observed in my cells (warranted gfp expression). Could you please suggest some reasons that I cound not find the gtf gene in 02.count/features.tsv.gz? 
![gtf_alignment](https://github.com/MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software/assets/16936914/33eb0310-4899-4174-927b-efb079aee30a)

Best regards,

Cheng

**Comment by lishuangshuang0616** [comment]
i need logfile in log dir.
find the reasons why genes are not annotated.

**Comment by zebrafishCC** [comment]
Sure, the attached 20240515.txt should be the logfile. The other file _bt_log seems to be not relevant and is not attached here.




-----Original Messages-----
From:lishuangshuang0616 ***@***.***>
Send time:Thursday, 05/16/2024 21:13:11
To: MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software ***@***.***>
Cc: Chengchen ***@***.***>, Author ***@***.***>
Subject: Re: [MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software] dnbc4tools rna run: Unable to find custom marker genes in the 02.count/features.tsv.gz file even they are appeared in the 01.data/final_sorted.bam (Issue #74)






i need logfile in log dir.
find the reasons why genes are not annotated.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>
/public/home/chengchen_gibh/miniconda3/envs/dnbc4tools/bin/dnbc4tools rna data --cDNAfastq1 nmpD4KO_mRNA_1.fq.gz --cDNAfastq2 nmpD4KO_mRNA_2.fq.gz --oligofastq1 nmpD4KO_oligo_1.fq.gz --oligofastq2 nmpD4KO_oligo_2.fq.gz --threads 8 --name nmpD4KO --chemistry auto --darkreaction auto --outdir /public/home/chengchen_gibh/huada/nmpD4KO --genomeDir /public/home/chengchen_gibh/GRC38ref/database --gtf /public/home/chengchen_gibh/GRC38ref/customHg38color.gtf --chrMT chrM
2024-05-15 14:15:41,493 - INFO - cDNAR1 fastq includes
2024-05-15 14:15:41,493 - INFO - 
2024-05-15 14:15:41,493 - INFO - /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO_mRNA_1.fq.gz
2024-05-15 14:15:41,493 - INFO - 
2024-05-15 14:15:41,493 - INFO - cDNAR2 fastq includes
2024-05-15 14:15:41,493 - INFO - 
2024-05-15 14:15:41,493 - INFO - /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO_mRNA_2.fq.gz
2024-05-15 14:15:41,493 - INFO - 
2024-05-15 14:15:41,493 - INFO - cDNA fastq config file /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/config/cellbarcode/scRNA_beads_darkReaction.json
2024-05-15 14:15:41,493 - INFO - 
2024-05-15 14:15:41,496 - INFO - olgioR1 fastq includes
2024-05-15 14:15:41,496 - INFO - 
2024-05-15 14:15:41,496 - INFO - ['/public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO_oligo_1.fq.gz']
2024-05-15 14:15:41,496 - INFO - 
2024-05-15 14:15:41,496 - INFO - olgioR2 fastq includes
2024-05-15 14:15:41,496 - INFO - 
2024-05-15 14:15:41,496 - INFO - ['/public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO_oligo_2.fq.gz']
2024-05-15 14:15:41,496 - INFO - 
2024-05-15 14:15:41,496 - INFO - oligo fastq config file /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/config/cellbarcode/scRNA_oligo_R2_noDarkReaction.json
2024-05-15 14:15:41,496 - INFO - 
2024-05-15 14:15:41,518 - data - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/parseFq /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/oligo_para
2024-05-15 14:15:41,519 - data - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/scStar --outSAMattributes singleCell --outSAMtype BAM Unsorted --genomeDir /public/home/chengchen_gibh/GRC38ref/database --outFileNamePrefix ../01.data/ --stParaFile /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/cDNA_para --outSAMmode NoQS --runThreadN 8 --limitOutSJcollapsed 10000000 --limitIObufferSize 350000000
2024-05-15 14:21:44,079 - data - INFO - mapped_dnbs: 582236

2024-05-15 15:54:50,674 - data - INFO - 2024-5-15  14:15:41 ..... started STAR run
May 15 14:15:41 ..... loading genome
May 15 14:16:36 ..... started mapping
mapped_dnbs: 2333315
May 15 15:54:38 ..... finished mapping
May 15 15:54:49 ..... finished successfully

2024-05-15 15:54:51,478 - data - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/Anno -I /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/Aligned.out.bam -a /public/home/chengchen_gibh/GRC38ref/customHg38color.gtf -L /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/cDNA.barcode.counts.txt -o /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data -c 8 -m chrM -B /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/config/cellbarcode/scRNA_beads_darkReaction.json --anno 1 --intron
2024-05-15 16:59:12,916 - data - INFO - 2024-05-15 15:55:38.344 W process: Multiple gene IDs for gene bfp: bfp -- skipping
2024-05-15 15:55:38.626 W process: Strand disagreement for gene TMSB15B -- skipping
2024-05-15 15:55:39.009 W process: Strand disagreement for gene LINC01238 -- skipping
memory before process: 1255
process reads: 10000000 time(s): 62
process reads: 20000000 time(s): 61
process reads: 30000000 time(s): 62
process reads: 40000000 time(s): 61
process reads: 50000000 time(s): 61
process reads: 60000000 time(s): 61
process reads: 70000000 time(s): 62
process reads: 80000000 time(s): 61
process reads: 90000000 time(s): 62
process reads: 100000000 time(s): 61
process reads: 110000000 time(s): 61
process reads: 120000000 time(s): 61
process reads: 130000000 time(s): 61
process reads: 140000000 time(s): 63
process reads: 150000000 time(s): 61
process reads: 160000000 time(s): 61
process reads: 170000000 time(s): 62
process reads: 180000000 time(s): 61
process reads: 190000000 time(s): 62
process reads: 200000000 time(s): 62
process reads: 210000000 time(s): 62
process reads: 220000000 time(s): 62
process reads: 230000000 time(s): 62
process reads: 240000000 time(s): 62
process reads: 250000000 time(s): 62
process reads: 260000000 time(s): 62
process reads: 270000000 time(s): 65
process reads: 280000000 time(s): 62
process reads: 290000000 time(s): 62
process reads: 300000000 time(s): 62
process reads: 310000000 time(s): 62
process reads: 320000000 time(s): 62
process reads: 330000000 time(s): 62
process reads: 340000000 time(s): 62
process reads: 350000000 time(s): 62
process reads: 360000000 time(s): 62
memory after collect annotation info: 7700 2276
memory after umi correction: 7899 309
process reads: 10000000 time(s): 20
process reads: 20000000 time(s): 22
process reads: 30000000 time(s): 23
process reads: 40000000 time(s): 24
process reads: 50000000 time(s): 24
process reads: 60000000 time(s): 25
process reads: 70000000 time(s): 25
process reads: 80000000 time(s): 26
process reads: 90000000 time(s): 26
process reads: 100000000 time(s): 27
process reads: 110000000 time(s): 27
process reads: 120000000 time(s): 27
process reads: 130000000 time(s): 28
process reads: 140000000 time(s): 27
process reads: 150000000 time(s): 28
process reads: 160000000 time(s): 27
process reads: 170000000 time(s): 28
process reads: 180000000 time(s): 28
process reads: 190000000 time(s): 29
process reads: 200000000 time(s): 29
process reads: 210000000 time(s): 29
process reads: 220000000 time(s): 29
process reads: 230000000 time(s): 29
process reads: 240000000 time(s): 30
process reads: 250000000 time(s): 30
process reads: 260000000 time(s): 30
process reads: 270000000 time(s): 30
process reads: 280000000 time(s): 30
process reads: 290000000 time(s): 31
process reads: 300000000 time(s): 30
process reads: 310000000 time(s): 31
process reads: 320000000 time(s): 31
process reads: 330000000 time(s): 31
process reads: 340000000 time(s): 31
process reads: 350000000 time(s): 31
process reads: 360000000 time(s): 31
memory after write bam: 27153 1039
memory after stat beads: 27352 21

2024-05-15 16:59:13,016 - data - INFO - Executing command: /public/home/chengchen_gibh/miniconda3/envs/dnbc4tools/bin/samtools sort -@ 8 /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/final.bam -o /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/final_sorted.bam
2024-05-15 17:08:20,694 - data - INFO - [bam_sort_core] merging from 23 files and 8 in-memory blocks...

2024-05-15 17:08:20,976 - INFO - Analyze oligo data to obtain the combination of cDNA cell barcodes and oligo index barcodes.
2024-05-15 17:08:20,977 - INFO - 
2024-05-15 17:08:21,410 - INFO - Read the whitelist and position config of oligo R2.
2024-05-15 17:08:21,410 - INFO - 
2024-05-15 17:09:15,934 - INFO - Filter the oligo index cell barcodes in oligo R2 fastq.
2024-05-15 17:09:16,071 - INFO - 
2024-05-15 17:10:43,698 - INFO - Complete the combined analysis of cDNA and oligo and generate /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/CB_UB_count.txt
2024-05-15 17:10:43,810 - INFO - 
/public/home/chengchen_gibh/miniconda3/envs/dnbc4tools/bin/dnbc4tools rna count --name nmpD4KO --calling_method emptydrops --expectcells 3000 --threads 8 --outdir /public/home/chengchen_gibh/huada/nmpD4KO
2024-05-15 17:11:52,693 - count - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/similarity -n 8 nmpD4KO /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/CB_UB_count.txt /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/beads.barcodes.umi100.txt /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/config/cellbarcode/oligo_type.txt /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/similarity.all.csv /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/similarity.droplet.csv /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/similarity.dropletfiltered.csv
2024-05-15 17:13:10,558 - count - INFO - Total 59982 beads
1000 beads have been processed
2000 beads have been processed
3000 beads have been processed
4000 beads have been processed
5000 beads have been processed
6000 beads have been processed
7000 beads have been processed
8000 beads have been processed
9000 beads have been processed
10000 beads have been processed
11000 beads have been processed
12000 beads have been processed
13000 beads have been processed
14000 beads have been processed
15000 beads have been processed
16000 beads have been processed
17000 beads have been processed
18000 beads have been processed
19000 beads have been processed
20000 beads have been processed
21000 beads have been processed
22000 beads have been processed
23000 beads have been processed
24000 beads have been processed
25000 beads have been processed
26000 beads have been processed
27000 beads have been processed
28000 beads have been processed
29000 beads have been processed
30000 beads have been processed
31000 beads have been processed
32000 beads have been processed
33000 beads have been processed
34000 beads have been processed
35000 beads have been processed
36000 beads have been processed
37000 beads have been processed
38000 beads have been processed
39000 beads have been processed
40000 beads have been processed
41000 beads have been processed
42000 beads have been processed
43000 beads have been processed
44000 beads have been processed
45000 beads have been processed
46000 beads have been processed
47000 beads have been processed
48000 beads have been processed
49000 beads have been processed
50000 beads have been processed
51000 beads have been processed
52000 beads have been processed
53000 beads have been processed
54000 beads have been processed
55000 beads have been processed
56000 beads have been processed
57000 beads have been processed
58000 beads have been processed
59000 beads have been processed
item cnt: 972701
bead count: 59982
Program costs: 78s

2024-05-15 17:13:10,558 - INFO - Merge beads within the same droplet based on correlation values.
2024-05-15 17:13:10,559 - INFO - 
2024-05-15 17:13:12,239 - INFO - Generate beads and merge corresponding cell id files /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/barcodeTranslate.txt.
2024-05-15 17:13:12,240 - INFO - 
2024-05-15 17:13:40,271 - count - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/tagAdd -n 8 -bam /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/final_sorted.bam -file /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/barcodeTranslate.hex.txt -out /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/anno_decon_sorted.bam -tag_check CB:Z: -tag_add DB:Z:
2024-05-15 17:28:25,688 - count - INFO - Program costs: 885s
input bam: /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/01.data/final_sorted.bam
file: /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/barcodeTranslate.hex.txt
output bam: /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/anno_decon_sorted.bam
tag check: CB:Z:
tag add: DB:Z:
nthreds: 8

2024-05-15 17:28:25,812 - count - INFO - Executing command: /public/home/chengchen_gibh/miniconda3/envs/dnbc4tools/bin/samtools index -@ 8 /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/anno_decon_sorted.bam
2024-05-15 17:29:26,312 - count - INFO - 
2024-05-15 17:29:26,486 - count - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/PISA count -one-hit -@ 8 -cb DB -anno-tag GN -umi UB -list /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/cell.id -outdir /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/raw_matrix /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/anno_decon_sorted.bam
2024-05-15 17:36:03,036 - count - INFO - [2024-05-15 17:36:00] [32mReal time: 392.714 sec; CPU: 1969.451 sec; Peak RSS: 28.857 GB.[0m

2024-05-15 17:36:03,347 - INFO - Run the cell calling step and read in the raw matrix
2024-05-15 17:36:03,351 - INFO - 
2024-05-15 17:39:05,499 - INFO - Number of distinct N supplied: 4085
2024-05-15 17:39:05,500 - INFO - 
2024-05-15 17:39:05,500 - INFO - Range of N: 6870
2024-05-15 17:39:05,500 - INFO - 
2024-05-15 17:39:05,500 - INFO - Number of features: 33889
2024-05-15 17:39:05,500 - INFO - 
2024-05-15 17:39:10,285 - INFO - .
2024-05-15 17:39:14,827 - INFO - .
2024-05-15 17:39:19,369 - INFO - .
2024-05-15 17:39:23,777 - INFO - .
2024-05-15 17:39:28,321 - INFO - .
2024-05-15 17:39:32,849 - INFO - .
2024-05-15 17:39:37,243 - INFO - .
2024-05-15 17:39:41,778 - INFO - .
2024-05-15 17:39:46,313 - INFO - .
2024-05-15 17:39:50,712 - INFO - .
2024-05-15 17:39:55,238 - INFO - .
2024-05-15 17:39:59,773 - INFO - .
2024-05-15 17:40:04,181 - INFO - .
2024-05-15 17:40:08,708 - INFO - .
2024-05-15 17:40:13,243 - INFO - .
2024-05-15 17:40:17,660 - INFO - .
2024-05-15 17:40:22,197 - INFO - .
2024-05-15 17:40:26,744 - INFO - .
2024-05-15 17:40:31,289 - INFO - .
2024-05-15 17:40:35,704 - INFO - .
2024-05-15 17:40:40,250 - INFO - .
2024-05-15 17:40:44,787 - INFO - .
2024-05-15 17:40:49,192 - INFO - .
2024-05-15 17:40:53,734 - INFO - .
2024-05-15 17:40:58,277 - INFO - .
2024-05-15 17:41:02,681 - INFO - .
2024-05-15 17:41:07,223 - INFO - .
2024-05-15 17:41:11,775 - INFO - .
2024-05-15 17:41:16,174 - INFO - .
2024-05-15 17:41:20,728 - INFO - .
2024-05-15 17:41:25,278 - INFO - .
2024-05-15 17:41:29,696 - INFO - .
2024-05-15 17:41:34,241 - INFO - .
2024-05-15 17:41:38,782 - INFO - .
2024-05-15 17:41:43,322 - INFO - .
2024-05-15 17:41:47,735 - INFO - .
2024-05-15 17:41:52,274 - INFO - .
2024-05-15 17:41:56,815 - INFO - .
2024-05-15 17:42:01,224 - INFO - .
2024-05-15 17:42:05,771 - INFO - .
2024-05-15 17:42:10,310 - INFO - .
2024-05-15 17:42:14,722 - INFO - .
2024-05-15 17:42:19,268 - INFO - .
2024-05-15 17:42:23,817 - INFO - .
2024-05-15 17:42:28,229 - INFO - .
2024-05-15 17:42:32,774 - INFO - .
2024-05-15 17:42:37,328 - INFO - .
2024-05-15 17:42:41,751 - INFO - .
2024-05-15 17:42:46,299 - INFO - .
2024-05-15 17:42:50,843 - INFO - .
2024-05-15 17:42:55,396 - INFO - .
2024-05-15 17:42:59,812 - INFO - .
2024-05-15 17:43:04,363 - INFO - .
2024-05-15 17:43:08,915 - INFO - .
2024-05-15 17:43:13,326 - INFO - .
2024-05-15 17:43:17,882 - INFO - .
2024-05-15 17:43:22,431 - INFO - .
2024-05-15 17:43:26,852 - INFO - .
2024-05-15 17:43:31,405 - INFO - .
2024-05-15 17:43:35,957 - INFO - .
2024-05-15 17:43:40,376 - INFO - .
2024-05-15 17:43:44,923 - INFO - .
2024-05-15 17:43:49,476 - INFO - .
2024-05-15 17:43:53,891 - INFO - .
2024-05-15 17:43:58,446 - INFO - .
2024-05-15 17:44:02,991 - INFO - .
2024-05-15 17:44:07,542 - INFO - .
2024-05-15 17:44:11,952 - INFO - .
2024-05-15 17:44:16,482 - INFO - .
2024-05-15 17:44:21,028 - INFO - .
2024-05-15 17:44:25,448 - INFO - .
2024-05-15 17:44:30,009 - INFO - .
2024-05-15 17:44:34,560 - INFO - .
2024-05-15 17:44:38,969 - INFO - .
2024-05-15 17:44:43,516 - INFO - .
2024-05-15 17:44:48,068 - INFO - .
2024-05-15 17:44:52,479 - INFO - .
2024-05-15 17:44:57,027 - INFO - .
2024-05-15 17:45:01,574 - INFO - .
2024-05-15 17:45:05,993 - INFO - .
2024-05-15 17:45:10,550 - INFO - .
2024-05-15 17:45:15,095 - INFO - .
2024-05-15 17:45:19,648 - INFO - .
2024-05-15 17:45:24,056 - INFO - .
2024-05-15 17:45:28,593 - INFO - .
2024-05-15 17:45:33,138 - INFO - .
2024-05-15 17:45:37,553 - INFO - .
2024-05-15 17:45:42,100 - INFO - .
2024-05-15 17:45:46,660 - INFO - .
2024-05-15 17:45:51,085 - INFO - .
2024-05-15 17:45:55,620 - INFO - .
2024-05-15 17:46:00,159 - INFO - .
2024-05-15 17:46:04,569 - INFO - .
2024-05-15 17:46:09,113 - INFO - .
2024-05-15 17:46:13,667 - INFO - .
2024-05-15 17:46:18,090 - INFO - .
2024-05-15 17:46:22,634 - INFO - .
2024-05-15 17:46:27,182 - INFO - .
2024-05-15 17:46:31,725 - INFO - .
2024-05-15 17:46:36,151 - INFO - .
2024-05-15 17:46:36,195 - INFO - 
2024-05-15 17:46:36,520 - INFO - Cell calling obtains 8114 number of cells.
2024-05-15 17:46:36,521 - INFO - 
2024-05-15 17:50:04,443 - count - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/PISA count -one-hit -@ 8 -cb DB -anno-tag GN -umi UB -list /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/beads_barcodes.txt -outdir /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/filter_matrix /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/anno_decon_sorted.bam
2024-05-15 17:53:15,098 - count - INFO - [2024-05-15 17:53:13] [32mReal time: 187.607 sec; CPU: 991.206 sec; Peak RSS: 18.300 GB.[0m

/public/home/chengchen_gibh/miniconda3/envs/dnbc4tools/bin/dnbc4tools rna analysis --name nmpD4KO --outdir /public/home/chengchen_gibh/huada/nmpD4KO --species Homo_sapiens --mtgenes /public/home/chengchen_gibh/GRC38ref/database/mtgene.list
Automatically set threshold at doublet score = 0.57
Detected doublet rate = 0.2%
Estimated detectable doublet fraction = 27.4%
Overall doublet rate:
	Expected   = 5.0%
	Estimated  = 0.7%
/public/home/chengchen_gibh/miniconda3/envs/dnbc4tools/bin/dnbc4tools rna report --name nmpD4KO --species Homo_sapiens --threads 8 --outdir /public/home/chengchen_gibh/huada/nmpD4KO
2024-05-15 18:15:08,420 - report - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/PISA count -one-hit -@ 8 -cb DB -ttype E,S -anno-tag GN -umi UB -list /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/beads_barcodes.txt -outdir /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/output/attachment/splice_matrix /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/output/anno_decon_sorted.bam
2024-05-15 18:18:12,239 - report - INFO - [2024-05-15 18:18:10] [32mReal time: 181.797 sec; CPU: 980.762 sec; Peak RSS: 17.556 GB.[0m

2024-05-15 18:18:12,363 - report - INFO - Executing command: /public/home/chengchen_gibh/.local/lib/python3.8/site-packages/dnbc4tools/software/PISA count -one-hit -@ 8 -cb DB -velo -anno-tag GN -umi UB -list /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/02.count/beads_barcodes.txt -outdir /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/output/attachment/RNAvelocity_matrix /public/home/chengchen_gibh/huada/nmpD4KO/nmpD4KO/output/anno_decon_sorted.bam
2024-05-15 18:21:41,369 - report - INFO - [2024-05-15 18:21:39] [32mReal time: 206.199 sec; CPU: 1051.987 sec; Peak RSS: 22.150 GB.[0m

**Comment by lishuangshuang0616** [comment]
I suggest renaming the chromosome names for **_mcherry_**, **_bfp_**, and _**gfp**_ to avoid any confusion with other information. For example, change them to **_chrmcherry_**, _**chrbfp**_, and _**chrgfp**_. Also, modify the gene_id and gene_name to **_GenMcherry_**, **_GenBfp_**, and **_GenGfp_**, respectively, to ensure they do not match the chromosome names. If this is urgent, you can analyze a subset of the cDNA raw data to check for any issues. From the screenshot, it seems that the gfp position should be able to annotate the gene.

**Comment by zebrafishCC** [comment]
Thanks for your suggestion! I changed the naming of custom genes in gtf file and fa file as suggested, however, I still could not find these genes in 02.count/raw_matrix/features.tsv.gz file. Of note,  although reads can map to custom GFP genes in the 01.data/final_sorted.bam, there is no GN:Z tag for these reads, while for other genes found in the features.tsv.gz file, there are GN:Z tags coming along with the corresponding reads.  An exemplary screenshot  of correct custom gtf file will be greatly appreciated.

**Comment by lishuangshuang0616** [comment]
Can you send me the alignment screenshots of these chromosomes in the final_sorted.bam file? And also the log file.

**Comment by lishuangshuang0616** [comment]
Why are sequence lengths inconsistent?

**Comment by zebrafishCC** [comment]
> Can you send me the alignment screenshots of these chromosomes in the final_sorted.bam file? And also the log file.

Sure. It seems no big difference to me when compared to the last results. 
![chrgfp_mapping](https://github.com/MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software/assets/16936914/53eb5195-ba4c-47d4-a274-acb1ea8bcc5d)
[20240517.txt](https://github.com/MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software/files/15359092/20240517.txt)

**Comment by zebrafishCC** [comment]
> Why are sequence lengths inconsistent?

Sorry, not sure which sequences. Do you mean that read sequences in the final_sorted.bam are inconsistent?

**Comment by lishuangshuang0616** [comment]
I noticed that the fragment lengths are inconsistent. Aren't they usually 100bp?

**Comment by lishuangshuang0616** [comment]
Let me confirm: after the modifications, did you rebuild the reference library using dnbc4tools rna mkref with the newly adjusted FASTA and GTF files? You changed the FASTA, and the GTF was also changed, right? I'll investigate this issue further. Thanks! @zebrafishCC

**Comment by zebrafishCC** [comment]
Thanks for your following! With the help of your colleague Wan, I found that the problem in my custom gtf. I just copied and pasted the three custom records from a previously successful annotation to the human official gtf file. However, this copy and paste step turned the tab(\t) between different columns of the gtf file into space, which leads to failure of gene annotation during the following steps. I am sorry for this mistake and thanks for your time to track this issue!

2024-05-20 15:41:58 "Shuangshuang Li" ***@***.***> 写道：

Let me confirm: after the modifications, did you rebuild the reference library using dnbc4tools rna mkref with the newly adjusted FASTA and GTF files? You changed the FASTA, and the GTF was also changed, right? I'll investigate this issue further. Thanks! @zebrafishCC
